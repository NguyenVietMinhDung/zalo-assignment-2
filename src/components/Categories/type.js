export type Name = {
  vi: string,
  en: string,
};

export type Product = {
  id: ?string,
  price: number,
  name: Name,
  img: string,
};

export type Category = {
  id: string,
  name: string,
  items: Array<Product>,
};

export type Props = {
  categories: Array<Category>
};
