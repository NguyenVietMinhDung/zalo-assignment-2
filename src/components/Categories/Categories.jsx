import React from 'react';
import './Categories.scss';
import Category from './Category';
import { Props } from './type';

const Categories = (props: Props) => {
  const { categories } = props;
  return (
    <div className="categories">
      {categories.map(category => <Category key={category.id} {...category} />)}
    </div>
  );
};

export default Categories;
