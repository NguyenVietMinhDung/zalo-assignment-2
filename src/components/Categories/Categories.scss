@import '../../styles/variables';

.categories {
	padding: 5px;

	background-color: $white-smoke;
}
