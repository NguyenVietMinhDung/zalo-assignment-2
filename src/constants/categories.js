export default [
  {
    id: 'other',
    name: 'Khác',
    items: [
      {
        id: 0,
        name: {
          vi: 'Phin Inox',
          en: 'Phin Inox',
        },
        price: 80000,
        img: '270_crop_PHININOX.png',
      },
      {
        id: 1,
        name: {
          vi: 'Ly Sứ Mosaic',
          en: 'Mosaic Mug',
        },
        price: 29000,
        img: '270_crop_LYSU.png',
      },
    ],
  },
  {
    id: 'bread',
    name: 'Bánh Mì',
    items: [
      {
        id: 0,
        name: {
          vi: 'Xíu Mại',
          en: 'Pork Meatball',
        },
        price: 19000,
        img: '270_crop_BMXIUMAI.png',
      },
      {
        id: 1,
        name: {
          vi: 'Chả Lụa Xá Xíu',
          en: 'Roast Pork & Pork Sausage',
        },
        price: 19000,
        img: '270_crop_BMCHALUAXAXIU.png',
      },
      {
        id: 2,
        name: {
          vi: 'Gà Xé Nước Tương',
          en: 'Roast Chicken',
        },
        price: 19000,
        img: '270_crop_BMGAXE.png',
      },
      {
        id: 3,
        name: {
          vi: 'Thịt Nướng',
          en: 'Grilled Pork',
        },
        price: 19000,
        img: '270_crop_BMTHITNUONG.png',
      },
    ],
  },
  {
    id: 'cake',
    name: 'Bánh Ngọt',
    items: [
      {
        id: 0,
        name: {
          vi: 'Bánh Mousse Cacao',
          en: 'Cocoa Mousse Cake',
        },
        price: 29000,
        img: '270_crop_MOUSSECACAO.png',
      },
      {
        id: 1,
        name: {
          vi: 'Bánh Mousse Đào',
          en: 'Peach Mousse Cake',
        },
        price: 29000,
        img: '270_crop_MOUSSEDAO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Bánh Sô-cô-la Highlands',
          en: 'Highlands Chocolate Bomb',
        },
        price: 29000,
        img: '270_crop_SOCOLAHL.png',
      },
    ],
  },
  {
    id: 'coffee',
    name: 'Cà Phê Đóng Gói',
    items: [
      {
        id: 0,
        name: {
          vi: 'Latte',
          en: 'Latte',
        },
        price: 60000,
        img: '270_crop_LATTE.png',
      },
      {
        id: 1,
        name: {
          vi: 'Cappuccino',
          en: 'Cappuccino',
        },
        price: 60000,
        img: '270_crop_CAPPUCINO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Americano',
          en: 'Americano',
        },
        price: 60000,
        img: '270_crop_AMERICANO.png',
      },
      {
        id: 3,
        name: {
          vi: 'Espresso',
          en: 'Espresso',
        },
        price: 60000,
        img: '270_crop_ESPRESSO.png',
      },
      {
        id: 4,
        name: {
          vi: 'Caramel Macchiato',
          en: 'Caramel Macchiato',
        },
        price: 60000,
        img: '270_crop_CARAMEL-MACCHIATO.png',
      },
    ],
  },
  {
    id: 'Tea',
    name: 'Trà',
    items: [
      {
        id: 0,
        name: {
          vi: 'Trà Thạch Vãi',
          en: 'Lychee Tea',
        },
        price: 39000,
        img: '270_crop_TRATHACHVAI_1.png',
      },
      {
        id: 1,
        name: {
          vi: 'Trà Thạch Đào',
          en: 'Peach Jelly Tea',
        },
        price: 39000,
        img: '270_crop_TRATHACHDAO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Trà Thanh Đào',
          en: 'Peach Lemongrass Tea',
        },
        price: 39000,
        img: '270_crop_TRATHANHDAO.png',
      },
      {
        id: 3,
        name: {
          vi: 'Trà Sen Vàng',
          en: 'Golden Lotus Tea',
        },
        price: 39000,
        img: '270_crop_TRASENVANG.png',
      },
    ],
  },
];
