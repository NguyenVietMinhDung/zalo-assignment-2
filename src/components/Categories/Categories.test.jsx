import React from 'react';
import { shallow } from 'enzyme';
import Categories from './Categories';
import categories from '../../constants/categories';

const defaultProps = {
  categories,
};

describe('<Categories />', () => {
  test('renders correctly', () => {
    const component = shallow(<Categories {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
