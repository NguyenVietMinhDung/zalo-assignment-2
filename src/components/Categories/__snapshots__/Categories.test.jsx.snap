// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> renders correctly 1`] = `
<div
  className="categories"
>
  <Category
    id="other"
    items={
      Array [
        Object {
          "id": 0,
          "img": "270_crop_PHININOX.png",
          "name": Object {
            "en": "Phin Inox",
            "vi": "Phin Inox",
          },
          "price": 80000,
        },
        Object {
          "id": 1,
          "img": "270_crop_LYSU.png",
          "name": Object {
            "en": "Mosaic Mug",
            "vi": "Ly Sứ Mosaic",
          },
          "price": 29000,
        },
      ]
    }
    key="other"
    name="Khác"
  />
  <Category
    id="bread"
    items={
      Array [
        Object {
          "id": 0,
          "img": "270_crop_BMXIUMAI.png",
          "name": Object {
            "en": "Pork Meatball",
            "vi": "Xíu Mại",
          },
          "price": 19000,
        },
        Object {
          "id": 1,
          "img": "270_crop_BMCHALUAXAXIU.png",
          "name": Object {
            "en": "Roast Pork & Pork Sausage",
            "vi": "Chả Lụa Xá Xíu",
          },
          "price": 19000,
        },
        Object {
          "id": 2,
          "img": "270_crop_BMGAXE.png",
          "name": Object {
            "en": "Roast Chicken",
            "vi": "Gà Xé Nước Tương",
          },
          "price": 19000,
        },
        Object {
          "id": 3,
          "img": "270_crop_BMTHITNUONG.png",
          "name": Object {
            "en": "Grilled Pork",
            "vi": "Thịt Nướng",
          },
          "price": 19000,
        },
      ]
    }
    key="bread"
    name="Bánh Mì"
  />
  <Category
    id="cake"
    items={
      Array [
        Object {
          "id": 0,
          "img": "270_crop_MOUSSECACAO.png",
          "name": Object {
            "en": "Cocoa Mousse Cake",
            "vi": "Bánh Mousse Cacao",
          },
          "price": 29000,
        },
        Object {
          "id": 1,
          "img": "270_crop_MOUSSEDAO.png",
          "name": Object {
            "en": "Peach Mousse Cake",
            "vi": "Bánh Mousse Đào",
          },
          "price": 29000,
        },
        Object {
          "id": 2,
          "img": "270_crop_SOCOLAHL.png",
          "name": Object {
            "en": "Highlands Chocolate Bomb",
            "vi": "Bánh Sô-cô-la Highlands",
          },
          "price": 29000,
        },
      ]
    }
    key="cake"
    name="Bánh Ngọt"
  />
  <Category
    id="coffee"
    items={
      Array [
        Object {
          "id": 0,
          "img": "270_crop_LATTE.png",
          "name": Object {
            "en": "Latte",
            "vi": "Latte",
          },
          "price": 60000,
        },
        Object {
          "id": 1,
          "img": "270_crop_CAPPUCINO.png",
          "name": Object {
            "en": "Cappuccino",
            "vi": "Cappuccino",
          },
          "price": 60000,
        },
        Object {
          "id": 2,
          "img": "270_crop_AMERICANO.png",
          "name": Object {
            "en": "Americano",
            "vi": "Americano",
          },
          "price": 60000,
        },
        Object {
          "id": 3,
          "img": "270_crop_ESPRESSO.png",
          "name": Object {
            "en": "Espresso",
            "vi": "Espresso",
          },
          "price": 60000,
        },
        Object {
          "id": 4,
          "img": "270_crop_CARAMEL-MACCHIATO.png",
          "name": Object {
            "en": "Caramel Macchiato",
            "vi": "Caramel Macchiato",
          },
          "price": 60000,
        },
      ]
    }
    key="coffee"
    name="Cà Phê Đóng Gói"
  />
  <Category
    id="Tea"
    items={
      Array [
        Object {
          "id": 0,
          "img": "270_crop_TRATHACHVAI_1.png",
          "name": Object {
            "en": "Lychee Tea",
            "vi": "Trà Thạch Vãi",
          },
          "price": 39000,
        },
        Object {
          "id": 1,
          "img": "270_crop_TRATHACHDAO.png",
          "name": Object {
            "en": "Peach Jelly Tea",
            "vi": "Trà Thạch Đào",
          },
          "price": 39000,
        },
        Object {
          "id": 2,
          "img": "270_crop_TRATHANHDAO.png",
          "name": Object {
            "en": "Peach Lemongrass Tea",
            "vi": "Trà Thanh Đào",
          },
          "price": 39000,
        },
        Object {
          "id": 3,
          "img": "270_crop_TRASENVANG.png",
          "name": Object {
            "en": "Golden Lotus Tea",
            "vi": "Trà Sen Vàng",
          },
          "price": 39000,
        },
      ]
    }
    key="Tea"
    name="Trà"
  />
</div>
`;
